python/pip_install/tools/wheel_installer/wheel.py

```diff
@@ -301,18 +383,39 @@ def _add(self, dep: str, platform: Optional[Platform]):
 
             self._select[p].add(dep)
 
-        if len(self._select[platform]) != 1:
+        if len(self._select[platform]) == 1:
+            # We are adding a new item to the select and we need to ensure that
+            # existing dependencies from less specialized platforms are propagated
+            # to the newly added dependency set.
+            for p, deps in self._select.items():
+                # Check if the existing platform overlaps with the given platform
+                if p == platform or platform not in p.all_specializations():
+                    continue
+
+                self._select[platform].update(self._select[p])
+
+    def _merge_to_common(self, dep):
```

Comment:
I can't think of a better name right now, any suggestions? It merges deps into the `self._deps` if it is present for all python versions that we target.

Reply:
`_maybe_add_common_dep()` ?

I like to use the "maybe" term when the condition isn't brief to describe. The term encourages readers pause and go look at the doc to understand the possible post-conditions.